README.md
# infoBlog
Plugin pour dotclear2, affiche toutes les informations à propos du blog.


changelog
v1.1 - 19/02/2018 - Pierre Van Glabeke
* cosmétique dc2.6
* localisation
* dc2.9 requis

_define.php
<?php 
# -- BEGIN LICENSE BLOCK ----------------------------------
# This file is part of infoBlog, a plugin for Dotclear.
# 
# Copyright (c) 2009 Tomtom
# http://blog.zenstyle.fr/
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
# -- END LICENSE BLOCK ------------------------------------

if (!defined('DC_RC_PATH')) { return; }

$this->registerModule(
		/* Name */			'infoBlog',
		/* Description */		'Display all informations about your blog',
		/* Author */			'Tomtom, Pierre Van Glabeke',
		/* Version */			'1.1',
	/* Properties */
	array(
		'permissions' => 'admin',
		'type' => 'plugin',
		'dc_min' => '2.9',
		'support' => 'https://forum.dotclear.org/viewforum.php?id=16',
		'details' => 'http://lab.dotclear.org/wiki/plugin/infoBlog/fr'
		)
);

_public.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
# This file is part of infoBlog, a plugin for Dotclear.
# 
# Copyright (c) 2009 Tomtom
# http://blog.zenstyle.fr/
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
# -- END LICENSE BLOCK ------------------------------------
if (!defined('DC_RC_PATH')) { return; }
class infoBlogPublic
{
	/**
	 * This function displays the infoBlog widget
	 *
	 * @param	w	Widget object
	 *
	 * @return	string
	 */
	public static function widget($w)
	{
		global $core;

		if ($w->offline)
			return;

		if (($w->homeonly == 1 && $core->url->type != 'default') ||
			($w->homeonly == 2 && $core->url->type == 'default')) {
			return;
		}

		$mask = '<li class="%1$s">%2$s</li>';
		
		$res = '';

		if ($w->displayentriesnumber) {
			$post_count = $core->blog->getPosts(array(),true)->f(0);
			$str_entries = ($post_count > 1) ? __('entries') : __('entry');
			$res .= sprintf($mask,'entries-number',$post_count.' '.$str_entries);
		}
		if ($w->displayselectedentriesnumber) {
			$post_selected_count = $core->blog->getPosts(array('post_selected' => 1),true)->f(0);
			$str_entries_selected = ($post_selected_count > 1) ? __('selected entries') : __('selected entry');
			$res .= sprintf($mask,'entries-number',$post_selected_count.' '.$str_entries_selected);
		}
		if ($w->displayentrieswaitingnumber) {
			$strReq = "SELECT count(P.post_id) FROM ".$core->prefix.
			"post P INNER JOIN ".$core->prefix.
			"user U ON U.user_id = P.user_id LEFT OUTER JOIN ".$core->prefix.
			"category C ON P.cat_id = C.cat_id WHERE P.blog_id = '".$core->blog->id.
			"' AND post_type = 'post' AND post_status = -2";
			$post_waiting_count = $core->con->select($strReq)->f(0);
			$str_entries_waiting = ($post_waiting_count > 1) ? __('entries waiting') : __('entry waiting');
			$res .= sprintf($mask,'entries-waiting-number',$post_waiting_count.' '.$str_entries_waiting);
		}
		if ($w->displayscheduledentriesnumber) {
			$strReq = "SELECT count(P.post_id) FROM ".$core->prefix.
			"post P INNER JOIN ".$core->prefix.
			"user U ON U.user_id = P.user_id LEFT OUTER JOIN ".$core->prefix.
			"category C ON P.cat_id = C.cat_id WHERE P.blog_id = '".$core->blog->id.
			"' AND post_type = 'post' AND post_status = -1";
			$post_scheduled_count = $core->con->select($strReq)->f(0);
			$str_entries_scheduled = ($post_scheduled_count > 1) ? __('scheduled entries') : __('scheduled entry');
			$res .= sprintf($mask,'entries-scheduled-number',$post_scheduled_count.' '.$str_entries_scheduled);
		}
		if ($w->displayofflineentriesnumber) {
			$strReq = "SELECT count(P.post_id) FROM ".$core->prefix.
			"post P INNER JOIN ".$core->prefix.
			"user U ON U.user_id = P.user_id LEFT OUTER JOIN ".$core->prefix.
			"category C ON P.cat_id = C.cat_id WHERE P.blog_id = '".$core->blog->id.
			"' AND post_type = 'post' AND post_status = -0";
			$post_offline_count = $core->con->select($strReq)->f(0);
			$str_entries_offline = ($post_offline_count > 1) ? __('offline entries') : __('offline entry');
			$res .= sprintf($mask,'entries-offline-number',$post_offline_count.' '.$str_entries_offline);
		}
		if ($w->displaycommentsnumber) {
			$comment_count = $core->blog->getComments(array(),true)->f(0);
			$str_comments = ($comment_count > 1) ? __('comments') : __('comment');
			$res .= sprintf($mask,'comments-number',$comment_count.' '.$str_comments);
		}
		if ($w->displaypingsnumber) {
			$ping_count = $core->blog->getComments(array('comment_trackback' => true),true)->f(0);
			$str_pings = ($comment_count > 1) ? __('pings') : __('ping');
			$res .= sprintf($mask,'pings-number',$ping_count.' '.$str_pings);
		}
		if ($w->displaystartblogdate) {
			$start = $core->blog->getPosts(array('order' => 'post_dt ASC','limit' => '1'));
			$str_start = dt::dt2str($core->blog->settings->system->date_format,$start->post_dt);
			$str_date = sprintf($w->displaystartblogdatetext,$str_start);
			if (!empty($str_date)) {
				$res .= sprintf($mask,'start-date',$str_date);
			}
		}
		if ($w->displayauthors) {
			$list = '';
			$users = $core->getUsers();
			while ($users->fetch()){
				$str = '<li>%1$s</li>';
				if ($users->user_url) {
					$str = '<li><a href="%2$s">%1$s</a></li>';
				}
				if ($w->displayauthorstats) {
					$post_count = $core->blog->getPosts(array('user_id' => $users->user_id),true)->f(0);
					$str_authors =
						rsExtPost::getAuthorCN($users).' - '.$post_count.' '.
						(($post_count > 1) ? __('entries') : __('entry'));
				}
				else {
					$str_authors = rsExtPost::getAuthorCN($users);
				}
				$list .= sprintf($str,$str_authors,$users->user_url);
			}
			if (!empty($list)) {
				$list = sprintf('<ul class="%1$s">%2$s</ul>','authors-list',$list);
				$res .= sprintf($mask,'authors',__('Authors:').$list);
			}
		}
		
		$res = !empty($res) ? '<ul>'.$res.'</ul>' : '';

    $res =
		($w->title ? $w->renderTitle(html::escapeHTML($w->title)) : '').
		$res;

		return $w->renderDiv($w->content_only,'info-blog '.$w->class,'',$res);
	}
}

_widgets.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
# This file is part of infoBlog, a plugin for Dotclear.
# 
# Copyright (c) 2009 Tomtom
# http://blog.zenstyle.fr/
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
# -- END LICENSE BLOCK ------------------------------------

if (!defined('DC_RC_PATH')) { return; }

$core->addBehavior('initWidgets',array('infoBlogWidgets','initWidgets'));

class infoBlogWidgets
{
	/**
	 * This function creates the infoBlog's widget object
	 *
	 * @param	w	Widget object
	 */
	public static function initWidgets($w)
	{
		$w->create('infoBlog',__('InfoBlog: informations about your blog'),array('infoBlogPublic','widget'),
			null,
			__('Display all informations about your blog'));
		$w->infoBlog->setting('title',__('Title:'),__('Informations about your blog'),'text');
		$w->infoBlog->setting('displayentriesnumber',__('Display entries number'),true,'check');
		$w->infoBlog->setting('displayselectedentriesnumber',__('Display selected entries number'),true,'check');
		$w->infoBlog->setting('displayentrieswaitingnumber',__('Display entries waiting\'s number'),true,'check');
		$w->infoBlog->setting('displayscheduledentriesnumber',__('Display scheduled entries number'),true,'check');
		$w->infoBlog->setting('displayofflineentriesnumber',__('Display offline entries number'),true,'check');
		$w->infoBlog->setting('displaycommentsnumber',__('Display comments number'),true,'check');
		$w->infoBlog->setting('displaypingsnumber',__('Display pings number'),true,'check');
		$w->infoBlog->setting('displaystartblogdate',__('Display start blog date'),true,'check');
		$w->infoBlog->setting('displaystartblogdatetext',__('Text to display start day:'),__('Blog start the %s'),'text');
		$w->infoBlog->setting('displayauthors',__('Display authors'),true,'check');
		$w->infoBlog->setting('displayauthorstats',__('Display author statistics'),true,'check');
		$w->infoBlog->setting('homeonly',__('Display on:'),0,'combo',
			array(
				__('All pages') => 0,
				__('Home page only') => 1,
				__('Except on home page') => 2
				)
		);
    $w->infoBlog->setting('content_only',__('Content only'),0,'check');
    $w->infoBlog->setting('class',__('CSS class:'),'');
		$w->infoBlog->setting('offline',__('Offline'),0,'check');
	}
}

locales/fr/main.lang.php
<?php
// Language: Français 
// Module: infoBlog - 1.1
// Date: 2018-02-19 13:50:54 
// Translated with dcTranslater - 2018.02.14 

#_public.php:45
$GLOBALS['__l10n']['selected entries'] = 'billets sélectionnés';

#_public.php:45
$GLOBALS['__l10n']['selected entry'] = 'billet sélectionné';

#_public.php:55
$GLOBALS['__l10n']['entries waiting'] = 'billets en attente';

#_public.php:55
$GLOBALS['__l10n']['entry waiting'] = 'billet en attente';

#_public.php:65
$GLOBALS['__l10n']['scheduled entries'] = 'billets programmés';

#_public.php:65
$GLOBALS['__l10n']['scheduled entry'] = 'billet programmé';

#_public.php:75
$GLOBALS['__l10n']['offline entries'] = 'billets hors ligne';

#_public.php:75
$GLOBALS['__l10n']['offline entry'] = 'billet hors ligne';

#_public.php:80
$GLOBALS['__l10n']['comments'] = 'commentaires';

#_public.php:85
$GLOBALS['__l10n']['pings'] = 'rétroliens';

#_public.php:85
$GLOBALS['__l10n']['ping'] = 'rétrolien';

#_public.php:117
$GLOBALS['__l10n']['Authors:'] = 'Auteurs :';

#_widgets.php:26
$GLOBALS['__l10n']['InfoBlog: informations about your blog'] = 'InfoBlog : informations sur votre blog';

#_widgets.php:28
$GLOBALS['__l10n']['Display all informations about your blog'] = 'Affiche toutes les informations sur votre blog';

#_widgets.php:29
$GLOBALS['__l10n']['Informations about your blog'] = 'Informations sur votre blog';

#_widgets.php:30
$GLOBALS['__l10n']['Display entries number'] = 'Afficher le nombre de billets';

#_widgets.php:31
$GLOBALS['__l10n']['Display selected entries number'] = 'Afficher le nombre de billets sélectionnés';

#_widgets.php:32
$GLOBALS['__l10n']['Display entries waiting\'s number'] = 'Afficher le nombre de billets en attente';

#_widgets.php:33
$GLOBALS['__l10n']['Display scheduled entries number'] = 'Afficher le nombre de billets programmés';

#_widgets.php:34
$GLOBALS['__l10n']['Display offline entries number'] = 'Afficher le nombre de billets hors ligne';

#_widgets.php:35
$GLOBALS['__l10n']['Display comments number'] = 'Afficher le nombre de commentaires';

#_widgets.php:36
$GLOBALS['__l10n']['Display pings number'] = 'Afficher le nombre de rétroliens';

#_widgets.php:37
$GLOBALS['__l10n']['Display start blog date'] = 'Afficher la date de création du blog';

#_widgets.php:38
$GLOBALS['__l10n']['Text to display start day:'] = 'Texte pour la date de création du blog :';

#_widgets.php:38
$GLOBALS['__l10n']['Blog start the %s'] = 'Blog créé le %s';

#_widgets.php:39
$GLOBALS['__l10n']['Display authors'] = 'Afficher les auteurs';

#_widgets.php:40
$GLOBALS['__l10n']['Display author statistics'] = 'Afficher les statistiques des auteurs';

